tests/file/test_handle.py

```diff
@@ -0,0 +1,153 @@
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Act
+        handle_generator = self.fs.list_handles(TEST_SHARE_NAME, recursive=True, max_results=1)
+
+        # Assert
+        self.assertIsNotNone(handle_generator.next_marker)
+        handles = list(handle_generator)
+        self._validate_handles(handles)
+
+        # Note down a handle that we saw
+        old_handle = handles[0]
+
+        # Continue listing
+        remaining_handles = list(self.fs.list_handles(TEST_SHARE_NAME, recursive=True, marker=handle_generator.next_marker))
+        self._validate_handles(handles)
+
+        # Make sure the old handle did not appear
+        # In other words, the marker worked
+        old_handle_not_present = all([old_handle.handle_id != handle.handle_id for handle in remaining_handles])
+        self.assertTrue(old_handle_not_present)
+
+    @record
+    def test_list_handles_on_directory(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Act
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, directory_name='wut', recursive=True))
+
+        # Assert
+        self._validate_handles(handles)
+
+    @record
+    def test_list_handles_on_file(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Act
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, directory_name='wut', file_name='bla.txt'))
+
+        # Assert
+        self._validate_handles(handles)
+
+    @record
+    def test_close_single_handle(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Arrange
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, recursive=True))
+        self._validate_handles(handles)
+        handle_id = handles[0].handle_id
+
+        # Act
+        num_closed = list(self.fs.close_handles(TEST_SHARE_NAME, handle_id=handle_id))
+
+        # Assert 1 handle has been closed
+        self.assertEqual(1, num_closed[0])
+
+    @record
+    def test_close_all_handle(self):
+        # don't run live, since the test set up was highly manual
+        # only run when recording, or playing back in CI
+        if not TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Arrange
+        handles = list(self.fs.list_handles(TEST_SHARE_NAME, recursive=True))
+        self._validate_handles(handles)
+
+        # Act
+        num_closed = list(self.fs.close_handles(TEST_SHARE_NAME, handle_id="*", recursive=True))
```

Comment:
Please test following continuation tokens, if I were to comment out line 264 of _deserialization.py by accident, this test wouldn't catch it :)

Reply:
That'd be too hard to test. The condition to trigger a marker is probably when there are more than 5000 open handles. There's currently no way to easily hold open handles, so it's not realistic to set up our unit tests to verify this. The main reason is that close_handles does not take in max_results, which allowed list_handles' marker to be tested.